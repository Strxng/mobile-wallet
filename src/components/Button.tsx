import React from "react";
import { TouchableOpacity, Text } from "react-native";
import FontAwesome from "@expo/vector-icons/FontAwesome";

interface ButtonProps {
  title: string;
  variant: "PRIMARY" | "SECONDARY";
  width: "50%" | "100%";
  iconName?: string;
  onPress?: () => void;
}

export const Button = ({
  title,
  variant,
  width,
  iconName,
  onPress = () => {},
}: ButtonProps) => {
  const color = variant === "PRIMARY" ? "#1c4ed8" : "#27272a";

  return (
    <TouchableOpacity
      className="p-3 h-14 justify-center items-center rounded-2xl flex flex-row"
      style={{ backgroundColor: color, gap: 10, width }}
      onPress={onPress}
    >
      {iconName && <FontAwesome size={20} name={iconName} color={"#FFF"} />}
      <Text className="text-white text-lg font-semibold">{title}</Text>
    </TouchableOpacity>
  );
};
