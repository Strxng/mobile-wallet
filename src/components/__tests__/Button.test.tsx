import React from "react";
import { Button } from "../Button";
import { render, cleanup } from "react-native-testing-library";

afterEach(cleanup);

describe("<Button />", () => {
  it("should match snapshot", () => {
    const rendered = render(
      <Button title={"teste"} variant={"PRIMARY"} width={"50%"} />
    ).toJSON();

    expect(rendered).toMatchSnapshot();
  });

  it("should render a button component with the title provided by props", () => {
    const buttonTitle = "titulo";
    const rendered = render(
      <Button title={buttonTitle} variant={"PRIMARY"} width={"50%"} />
    );

    const renderedButtonText = rendered.getByTestId("button-text");
    expect(renderedButtonText.props.children).toEqual(buttonTitle);
  });
});
