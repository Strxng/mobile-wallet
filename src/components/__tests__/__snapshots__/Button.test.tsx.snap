// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#1c4ed8",
      "borderBottomLeftRadius": 16,
      "borderBottomRightRadius": 16,
      "borderTopLeftRadius": 16,
      "borderTopRightRadius": 16,
      "display": "flex",
      "flexDirection": "row",
      "gap": 10,
      "height": 56,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
      "width": "50%",
    }
  }
  testID="button"
>
  <Text
    style={
      [
        {
          "color": "#fff",
        },
        {
          "fontSize": 18,
          "lineHeight": 28,
        },
        {
          "fontWeight": "600",
        },
      ]
    }
    testID="button-text"
  >
    teste
  </Text>
</View>
`;
